scripts/util.py

```diff
@@ -152,6 +155,61 @@
     None
 ] = lambda self, data: yaml.representer.SafeRepresenter.represent_str(self, str(data))
 
+with open(PROVIDERS_FILE) as stream:
+    try:
+        providers = yaml.safe_load(stream)
+    except yaml.YAMLError as exc:
+        providers = None
+        print(exc)
+prv = NSDict(providers)
+
+universe_domain = "googleapis.com"
+universe_sa_credentials = None
+endpoint_versions = None
+if prv.universe_information:
+    d, c = "domain", "sa_credentials"
+    if d in prv.universe_information and prv.universe_information[d]:
+        universe_domain = prv.universe_information[d]
+        # Only use credentials if domain is different
+        if c in prv.universe_information and prv.universe_information[c]:
+            universe_sa_credentials = "universe_sa_credentials.json"
+            with open("universe_sa_credentials.json", "w") as f:
+                f.write(prv.universe_information[c])
+if prv.endpoint_versions:
+    endpoint_versions = prv.endpoint_versions
+
+
+class ApiEndpoint(Enum):
+    COMPUTE = "compute"
+    BQ = "bq"
+    STORAGE = "storage"
+    TPU = "tpu"
+    SECRET = "secret_manager"
+
+
+def create_client_options(api: ApiEndpoint = None):
+    """Create client options for cloud endpoints"""
+    ep = None
+    if (
+        endpoint_versions
+        and api
+        and api.value in endpoint_versions
+        and endpoint_versions[api.value]
+    ):
+        ep = f"https://{api.value}.{universe_domain}/"
+        if api == ApiEndpoint.STORAGE:
+            os.environ["_API_VERSION_OVERRIDE_ENV_VAR"] = endpoint_versions[api.value]
+        else:
+            ep += f"{endpoint_versions[api.value]}"
```

Comment:
Why do you wrap it in string, isn't value of `endpoint_versions[api.value]` a string already?

Reply:
There was supposed to be a `/` at the end of that string

Reply:
Done.